subprojects/core/src/main/java/org/gradle/internal/classpath/intercept/DefaultCallSiteDecorator.java

```diff
@@ -270,6 +267,19 @@ private void restoreCallSiteArrayEntry() {
             array.array[index] = this;
         }
 
+        @Override
+        @Nullable
+        public Object callGetProperty(Object receiver) throws Throwable {
+            CallInterceptor interceptor = resolveCallInterceptor(InterceptScope.readsOfPropertiesNamed(getName()));
+            if (interceptor != null) {
+                return interceptor.intercept(
+                    new InvocationImpl<>(receiver, new Object[0], () -> maybeInstrumentedDynamicCall(CallStrategy.CALL_GET_PROPERTY, receiver, null)),
+                    callSiteOwnerClassName()
+                );
+            }
+            return maybeInstrumentedDynamicCall(CallStrategy.CALL_GET_PROPERTY, receiver, null);
```

Comment:
Is it the case that Groovy now generates `callGetProperty` for non-dynamic (statically known) property access? An example would be interesting. I don't mind handling it this way even if it doesn't, I'm just curious.

Reply:
Copied from Slack, for posterity:
> The specific behavior is that when you use a closure and use implicit receivers, e.g. `{ foo }` or `{ foo = "bar" }` or `{ foo() }`, the Groovy compiler no longer selects the `GroovyObject` variant of these calls (e.g. `InvokerHelper#getGroovyObjectProperty`). Instead, it calls the normal version which can forward to the same code path if the object is found to be an `instanceof GroovyObject` at runtime (e.g. `InvokerHelper#getProperty`).